src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -145,6 +145,44 @@ export const migrations = {
       } catch (e) {
         throw new Error(`Failure attempting to migrate saved object '${doc.attributes.title}' - ${e}`);
       }
+    },
+    '7.1.0': doc => {
+      // [TSVB] Migrate percentile-rank aggregation (value -> values)
+      const migratePercentileRankAggregation = doc => {
+        const visStateJSON = get(doc, 'attributes.visState');
+        let visState;
+
+        if (visStateJSON) {
+          try {
+            visState = JSON.parse(visStateJSON);
+          } catch (e) {
+            // Let it go, the data is invalid and we'll leave it as is
+          }
+          if (visState && visState.type === 'metrics') {
+            const series = get(visState, 'params.series') || [];
+
+            series.forEach(part => {
+              (part.metrics || []).forEach(metric => {
+                if (metric.type === 'percentile_rank' && has(metric, 'value')) {
```

Comment:
I wouldn't pull in a lodash function just for this. JS has `metric.hasOwnProperty('value')` which I think should be fine here. Other than this, I think this is good enough to ship.